src/pyenphase/envoy.py

```diff
@@ -273,3 +275,38 @@ async def go_off_grid(self) -> dict[str, Any]:
                 "This feature is not available on this Envoy."
             )
         return await self._json_request(URL_GRID_RELAY, {"mains_admin_state": "open"})
+
+    async def update_dry_contact(self, data: dict[str, Any]) -> dict[str, Any]:
+        """Update settings for an Enpower dry contact relay."""
+        # All settings for the relay must be sent in the POST or it may crash the Envoy
+
+        if not self.supported_features & SupportedFeatures.ENPOWER:
+            raise EnvoyFeatureNotAvailable(
+                "This feature is not available on this Envoy."
+            )
+
+        if "id" not in data.keys():
+            raise ValueError("You must specify the dry contact ID in the data object.")
+
+        id = data["id"]
+
+        # Get the current settings for the relay from EnvoyData and merge with the new settings
+        if not self.data:
+            raise ValueError(
+                "Tried to set dry contact settings before the Envoy was queried."
+            )
+        settings = self.data.dry_contact_settings[id]
+
+        for field in fields(settings):
+            setting = field.name
+            if setting in data:
+                setattr(settings, setting, data.pop(setting))
+
+        # If there is still information in data, it is invalid
+        if data:
+            _LOGGER.warning("Invalid data provided for dry contact relay: %s", data)
+
+        # Massage the settings into the format the Envoy API expects
+        new_settings = {"dry_contacts": settings.to_api()}
+
+        return await self._json_request(URL_DRY_CONTACT_SETTINGS, new_settings)
```

Comment:
```suggestion
    async def update_dry_contact(self, new_data: dict[str, Any]) -> dict[str, Any]:
        """Update settings for an Enpower dry contact relay."""
        # All settings for the relay must be sent in the POST or it may crash the Envoy
        if not self.supported_features & SupportedFeatures.ENPOWER:
            raise EnvoyFeatureNotAvailable(
                "This feature is not available on this Envoy."
            )
        if not (id_ := new_data.get("id")):
            raise ValueError("You must specify the dry contact ID in the data object.")
        # Get the current settings for the relay from EnvoyData and merge with the new settings
        if not (current_data := self.data):
            raise ValueError(
                "Tried to set dry contact settings before the Envoy was queried."
            )

        current_model = current_data.dry_contact_settings[id_]
        new_model = replace(current_model, **new_data)

        # Massage the settings into the format the Envoy API expects
        return await self._json_request(
            URL_DRY_CONTACT_SETTINGS, {"dry_contacts": new_model.to_api()}
        )
```

Reply:
I have no way to test the above

Reply:
Tested and works on my system. Much cleaner too 🦭